### ISSUE TITLE
Assertion Failed when opening ARM Templates

### ISSUE THREAD

**Issue by glloyd2010f** [microsoft/vscode-azurearmtools#441]
Does this occur consistently? Yes
Repro steps:
1. Save work in VS Code
2. Shutdown VS Code
3. Open ARM json File with VS Code

Additional information, this is happening after roughly one day after re-installing VS Code in the system installer format.

Action: reportDeploymentTemplateErrors
Error type: ERR_ASSERTION
Error Message: Assertion failed

Version: 0.8.3
OS: win32
OS Release: 10.0.17134
Product: Visual Studio Code
Product Version: 1.41.1
Language: en

<details>
<summary>Call Stack</summary>

```
Object.s [as assert] extension.bundle.js:1:70135
ObjectValue.getPropertyValueFromStack extension.bundle.js:1:103821
UndefinedVariablePropertyVisitor.visitPropertyAccess extension.bundle.js:127:143326
PropertyAccess.accept extension.bundle.js:1:120462
UndefinedVariablePropertyVisitor.visitFunctionCall extension.bundle.js:1:122651
FunctionCallValue.accept extension.bundle.js:1:119354
Function.visit extension.bundle.js:127:143787
GenericStringVisitor.onStringValue extension.bundle.js:127:111195
GenericStringVisitor.visitStringValue extension.bundle.js:218:778250
StringValue.accept extension.bundle.js:1:105619
GenericStringVisitor.visitProperty extension.bundle.js:1:110634
Property.accept extension.bundle.js:1:104371
GenericStringVisitor.visitObjectValue extension.bundle.js:1:110767
ObjectValue.accept extension.bundle.js:1:103923
GenericStringVisitor.visitArrayValue extension.bundle.js:1:110844
ArrayValue.accept extension.bundle.js:1:104792
GenericStringVisitor.visitProperty extension.bundle.js:1:110634
Property.accept extension.bundle.js:1:104371
GenericStringVisitor.visitObjectValue extension.bundle.js:1:110767
ObjectValue.accept extension.bundle.js:1:103923
GenericStringVisitor.visitArrayValue extension.bundle.js:1:110844
ArrayValue.accept extension.bundle.js:1:104792
GenericStringVisitor.visitProperty extension.bundle.js:1:110634
Property.accept extension.bundle.js:1:104371
GenericStringVisitor.visitObjectValue extension.bundle.js:1:110767
ObjectValue.accept extension.bundle.js:1:103923
GenericStringVisitor.visitArrayValue extension.bundle.js:1:110844
ArrayValue.accept extension.bundle.js:1:104792
GenericStringVisitor.visitProperty extension.bundle.js:1:110634
Property.accept extension.bundle.js:1:104371
GenericStringVisitor.visitObjectValue extension.bundle.js:1:110767
ObjectValue.accept extension.bundle.js:1:103923
Function.visit extension.bundle.js:218:778215
DeploymentTemplate.visitAllReachableStringValues extension.bundle.js:127:116930
DeploymentTemplate.<anonymous> extension.bundle.js:127:110658
extension.bundle.js:127:108290extension.bundle.js:127:108290
n extension.bundle.js:127:108038
extension.bundle.js:127:110560extension.bundle.js:127:110560
DeploymentTemplate.<anonymous> extension.bundle.js:127:110541
extension.bundle.js:127:108290extension.bundle.js:127:108290
n extension.bundle.js:127:108038
extension.bundle.js:127:110501extension.bundle.js:127:110501
CachedPromise.getOrCachePromise extension.bundle.js:127:117116
DeploymentTemplate.get errorsPromise [as errorsPromise] extension.bundle.js:127:110479
AzureRMTools.<anonymous> extension.bundle.js:127:180275
extension.bundle.js:127:176093extension.bundle.js:127:176093
n extension.bundle.js:127:175841
AzureRMTools.reportTemplateOpenedTelemetry extension.bundle.js:127:180123
extension.bundle.js:127:179609extension.bundle.js:127:179609
Object.t.callWithTelemetryAndErrorHandlingSync extension.bundle.js:1:218652
AzureRMTools.updateDeploymentTemplate extension.bundle.js:127:178812
AzureRMTools.onDocumentOpened extension.bundle.js:127:191674
l.fire extensionHostProcess.js:46:913
t.ExtHostDocuments.$acceptModelModeChanged extensionHostProcess.js:558:296
p._doInvokeHandler extensionHostProcess.js:663:275
p._invokeHandler extensionHostProcess.js:662:971
p._receiveRequest extensionHostProcess.js:661:639
p._receiveOneMessage extensionHostProcess.js:660:468
extensionHostProcess.js:658:691extensionHostProcess.js:658:691
l.fire extensionHostProcess.js:46:893
_.fire extensionHostProcess.js:189:274
extensionHostProcess.js:820:285extensionHostProcess.js:820:285
l.fire extensionHostProcess.js:46:893
_.fire extensionHostProcess.js:189:274
t.PersistentProtocol._receiveMessage extensionHostProcess.js:193:629
extensionHostProcess.js:190:824extensionHostProcess.js:190:824
l.fire extensionHostProcess.js:46:893
p.acceptChunk extensionHostProcess.js:186:737
extensionHostProcess.js:186:89extensionHostProcess.js:186:89
Socket.t extensionHostProcess.js:195:68
Socket.emit events.js:200:13
addChunk _stream_readable.js:294:12
readableAddChunk _stream_readable.js:275:11
Socket.Readable.push _stream_readable.js:210:10
Pipe.onStreamRead stream_base_commons.js:166:17
```

</details>

**Comment by StephenWeatherford** [comment]
public getPropertyValueFromStack(propertyNameStack: string[]): Value | null {
        // tslint:disable-next-line:no-this-assignment
        let result: Value | null = <Value | null>this;

        while (result && propertyNameStack.length > 0) {
            const objectValue: ObjectValue | null = asObjectValue(result);

            // We only handle evaluating properties in objects (e.g. not arrays)
            if (objectValue) {
                const propertyName = propertyNameStack.pop();
                result = propertyName ? objectValue.getPropertyValue(propertyName) : null;
            } else {
                assert(false);

**Comment by StephenWeatherford** [comment]
Two scenarios:
1) params:

 "parameters": {
                        "location": {
                            "type": "object",
                            "defaultValue": {
                                "a": [ // array inside an object
                                    1
                                ]
                            }
                        }
                    },
                    "outputs": {
                        "output1": {
                            "type": "bool",
                            "value": "[parameters('location').a.b.]"  <<< COMPLETE AFTER a.b.
                        }
                    }

2) variables (just open the file):

 "variables": {
                        "v1": {
                            "a": [
                                1
                            ]
                        }
                    },
                    "outputs": {
                        "output1": {
                            "type": "bool",
                            "value": "[variables('v1').a.b.c]"
                        }
                    }

**Comment by StephenWeatherford** [comment]
FYI, the fix for this bug has just been released in v0.8.4
